src/libDirectoryService/DirectoryService.cpp

```diff
@@ -1057,8 +1061,10 @@ uint8_t DirectoryService::CalculateNewDifficultyCore(uint8_t currentDifficulty,
     adjustment = -MAX_ADJUST_STEP;
   }
 
-  return std::max((uint8_t)(adjustment + currentDifficulty),
-                  (uint8_t)(minDifficulty));
+  uint16_t newDifficulty = (uint16_t)adjustment + (uint16_t)currentDifficulty;
```

Comment:
(uint16_t)adjustment shouldm't be uint?

Reply:
Also, adjustment can be -ve `adjustment = -MAX_ADJUST_STEP;`
 `(uint16_t)adjustment` can give unexpected result.

Reply:
Any issue with just keeping the original code for the return value?

Reply:
The original code cannot check maximum value, i will handle this case